src/dsp_tools/utils/excel_to_json/properties.py

```diff
@@ -0,0 +1,457 @@
+) -> str:
+    """
+    This function takes a list of properties, which are formatted as a dictionary and an error which was raised by the
+    calling function. It then searches which exact location of the property caused the error. The results are added
+    to a string which is used in the error that is raised in the calling function.
+
+    Args:
+        list_properties: List of properties
+        excel_filename: Name of the Excel file
+        validation_error: The error from the calling function
+
+    Returns:
+        A string which is used in the Error message that contains detailed information about the problem
+    """
+    err_msg_list = [f"The 'properties' section defined in the Excel file '{excel_filename}' did not pass validation."]
+    json_path_to_property = regex.search(r"^\$\[(\d+)\]", validation_error.json_path)
+    if json_path_to_property:
+        # fmt: off
+        wrong_property_name = (
+            jsonpath_ng.ext.parse(json_path_to_property.group(0))
+            .find(list_properties)[0]
+            .value["name"]
+        )
+        # fmt: on
+        excel_row = int(json_path_to_property.group(1)) + 2
+        err_msg_list.append(f"The problematic property is '{wrong_property_name}' in Excel row {excel_row}.")
+        affected_field = regex.search(
+            r"name|labels|comments|super|subject|object|gui_element|gui_attributes",
+            validation_error.json_path,
+        )
+        if affected_field:
+            err_msg_list.append(
+                f"The problem is that the column '{affected_field.group(0)}' has an invalid value: "
+                f"{validation_error.message}"
+            )
+    else:
+        err_msg_list.append(
+            f"The error message is: {validation_error.message}\n" f"The error occurred at {validation_error.json_path}"
+        )
+    return "\n".join(err_msg_list)
+
+
+def _validate_properties(
+    properties_list: list[dict[str, Any]],
+    excelfile: str,
+) -> bool:
+    """
+    This function checks if the "properties" section of a JSON project file is valid, according to the JSON schema.
+
+    Args:
+        properties_list: the "properties" section of a JSON project as a list of dicts
+        excelfile: path to the Excel file containing the properties
+
+    Raises:
+        UserError: if the validation fails
+
+    Returns:
+        True if the "properties" section passed validation
+    """
+    with importlib.resources.files("dsp_tools").joinpath("resources/schema/properties-only.json").open(
+        encoding="utf-8"
+    ) as schema_file:
+        properties_schema = json.load(schema_file)
+    try:
+        jsonschema.validate(instance=properties_list, schema=properties_schema)
+    except jsonschema.ValidationError as err:
+        err_msg = _search_validation_error(
+            list_properties=properties_list, excel_filename=excelfile, validation_error=err
```

Comment:
I tend to leave variable names as they are. Then I can search for `list_properties` in the entire file, and I find every occurrence. But if they are renamed to `list_properties` inside this function, I miss the ones inside this functions.

Reply:
I understand this point but in some cases this caused bugs for me that were really hard to find as a modification in a variable in a called function also changed the variable in the calling function. I started to do this as an easy way to prevent such mistakes. Only variables which have the same value everywhere were called the same. So in the excel case I'll rename it. In the other I will have a think...

Reply:
Are you talking about unwanted variable modifications like this?

```
def hidden_list_changer(some_list: list[int]) -> None:
    some_list.append(4)
    return None

my_list = [1,2,3]
hidden_list_changer(some_list=my_list)
print(my_list)
[1, 2, 3, 4]
```

In this case, variable names don't protect you! 

Or do you mean something else?

Reply:
Discussed below.